File: test/scripts/04-memory-allocation/incorrect.js
const N = 500_000;
const DURATION_SEC = parseInt(process.env.DURATION_SEC || '300', 10);
let laps = 0;
const start = Date.now();
console.log('INCORRECT: Many allocations');
function run() {
  if (Date.now() - start >= DURATION_SEC * 1000) {
    console.log('Stopped:', laps, 'laps');
    return;
  }
  const items = [];
  for (let i = 0; i < N; i++) items.push({ id: i, value: String(i).repeat(10) });
  laps++;
  setImmediate(run);
}
run();
process.on('SIGINT', () => process.exit(0));

File: test/scripts/04-memory-allocation/incorrect.py
#!/usr/bin/env python3
"""
INCORRECT: Many allocations, no pooling - GC pressure.
Resume: "Use Memory Pooling"
"""
import os
import signal
import time
N = 500_000
DURATION_SEC = int(os.environ.get("DURATION_SEC", "300"))
stop = False

def main():
    global stop
    print("INCORRECT: Excessive allocations")
    signal.signal(signal.SIGINT, lambda s, f: globals().__setitem__('stop', True))
    signal.signal(signal.SIGTERM, lambda s, f: globals().__setitem__('stop', True))
    start = time.perf_counter()
    laps = 0
    while not stop and (time.perf_counter() - start) < DURATION_SEC:
        items = []
        for i in range(N):
            items.append({"id": i, "value": str(i) * 10})
        laps += 1
    print(f"Stopped: {laps} laps (fix: pooling)")

if __name__ == "__main__":
    main()
